mesonbuild/modules/gnome.py

```diff
@@ -890,6 +890,8 @@ def gdbus_codegen(self, state, args, kwargs):
             cmd += ['--c-namespace', kwargs.pop('namespace')]
         if kwargs.get('object_manager', False):
             cmd += ['--c-generate-object-manager']
+        if 'autocleanup' in kwargs:
+            cmd += ['--c-generate-autocleanup', kwargs.pop('autocleanup')]
```

Comment:
Please add an `else` here which will add `['--c-generate-autocleanup', 'objects']` to `cmd`. We should turn it on by default. Also:

1. Please add a release note snippet, see `docs/markdown/snippets` for examples
2. Update the Gnome-module doc to say that the values can be `none`, `objects`, `all`, and that the default is `objects`
3. Validate the value passed via the kwarg and error out if it is invalid